src/tlc/pipeline/parser/joining-json-parser.ts

```diff
@@ -0,0 +1,47 @@
+import { Observable } from "rxjs";
+import Parser from "@app/pipeline/parser/index";
+import log from "@app/util/simple-logger";
+
+const newLineCharacter = "\n";
+
+/**
+ * Parser implementation that can convert multiple corresponding string objects into a single JSON document. This behavior
+ * is achieved by maintaining a local "buffer", and awaiting the corresponding pieces, then joining them together.
+ * Therefore, this parser must not be shared among different log streams, each log pipeline should have its own instance.
+ */
+export default class JoiningJsonParser implements Parser<string, object> {
+
+    private buffer: string[];
+
+    private constructor() {
+        this.buffer = [];
+    }
+
+    /**
+     * Returns a prototype instance of the JoiningJsonParser.
+     */
+    public static getInstance(): JoiningJsonParser {
+        return new JoiningJsonParser();
+    }
+
+    parse(inputData: string): Observable<object> {
+
+        return new Observable(subscriber => {
+
+            this.buffer.push(inputData);
+            if (inputData.endsWith(newLineCharacter)) {
+                this.buffer.join("")
+                    .trim()
```

Comment:
I'm not sure if that's intended, but I've just figured out that trimming has an additional benefit - if the input ends with a "\n", then the split will produce an empty string at the end of the array. However, since it's trimmed, it won't.

Reply:
Yep, I used a similar logic in Domino for processing streaming data from Docker Engine, and it was needed for some reason. I'm not exactly sure anymore why, but I think the data lines were having some extra white spaces, that needed to be removed. Same source, same precautions. :)